✨ feat: creates root key if not present

pallabpain opened the pull request:
### Description
This commit updates the initializer to create the root key if not present given `createKeyIfNotPresent` is set to true.

### Testing
The test runs on an actual `etcd3` cluster.
```
/usr/local/go/bin/go tool test2json -t /home/pallab/.cache/JetBrains/GoLand2023.1/tmp/GoLand/___adapter_test_go.test -test.v -test.paniconexit0 -test.run ^\QTestAdapters\E$
=== RUN   TestAdapters
    adapter_test.go:38: Test pass
    adapter_test.go:45: Test pass
    adapter_test.go:59: Test pass
    adapter_test.go:38: Test pass
    adapter_test.go:45: Test pass
    adapter_test.go:84: Test pass
    adapter_test.go:95: Test pass
    adapter_test.go:100: Test pass
    adapter_test.go:106: Test pass
--- PASS: TestAdapters (0.02s)
PASS
```

pallabpain committed:
✨ feat: creates root key if not present

This commit updates the initializer to create the root key if not
present given `createKeyIfNotPresent` is set to true.

pallabpain committed:
🔧 chore: updates go.mod

ankitrgadiya commented:
I'd recommend not including non-related changes like Github Actions / README updates in feature or bugfix PRs. The urgency of feature and bugfix is different from those changes. As a reviewer, this increases the scope of immediate changes I need to check. Additionally, there might be genuine comments/concerns with respect to those changes that I feel oblized to ignore because I don't want to hold off the feature or bugfix because of those changes.

pallabpain commented:
> I'll recommend not increasing the scope of PRs. Please add the Github Actions changes in a separate PR from now on. Let's keep the PRs small and tight.

Dropped the last three commits and moved them to a separate PR. https://github.com/rapyuta-robotics/etcd-adapter/pull/7

ankitrgadiya commented on adapter.go:
```diff
@@ -52,24 +52,26 @@ type CasbinRule struct {
 
-func NewAdapter(etcdEndpoints []string, key string) *Adapter {
-	return newAdapter(etcdEndpoints, key)
+func NewAdapter(etcdEndpoints []string, key string, createKeyIfNotPresent bool) *Adapter {
```
Why are we accepting this option? I feel like it makes sense to always do it as it is an idempotent operation and the only place we are using it will set it to true.

pallabpain replied:
Makes sense to make this implicit.

pallabpain committed:
fix: removes createIfKeyNotPresent flag

pallabpain committed:
fix: removes unused field from Adapter struct
